### PR TITLE
[WM-2414] clone recovery

### PR DESCRIPTION
**Pull request by mspector**

### CODE REVIEW

File: service/build.gradle [hunk 1]
```diff
@@ -74,6 +74,9 @@ dependencies {
     testImplementation('au.com.dius.pact.provider:junit5spring:4.3.19')
     testImplementation('au.com.dius.pact.consumer:junit5:4.3.19')
     testImplementation("au.com.dius.pact.consumer:junit:4.3.19")
+    testImplementation('org.testcontainers:junit-jupiter')
+    testImplementation('org.testcontainers:postgresql')
+    testImplementation('org.springframework.boot:spring-boot-testcontainers:3.2.3')
```
Comment (cjllanwarne) [inline service/build.gradle:79]: Could we extract testcontainers into a separate PR. It'd be nice to remove the noise from the clone recovery changes AND make the PRs more targeted and easier to find if we want to reference them in the future

File: service/src/test/java/bio/terra/cbas/dao/TestMethodDao.java [hunk 2]
```diff
@@ -50,22 +58,40 @@ class TestMethodDao {
...
+          .withDatabaseName("test_db")
+          .withUsername("test_user")
+          .withPassword("test_password");
+
+  @DynamicPropertySource
+  static void postgresProperties(DynamicPropertyRegistry registry) {
+    registry.add("spring.datasource.jdbc-url", postgres::getJdbcUrl);
+    registry.add("spring.datasource.username", postgres::getUsername);
+    registry.add("spring.datasource.password", postgres::getPassword);
+  }
```
Comment (cjllanwarne) [inline service/src/test/java/bio/terra/cbas/dao/TestMethodDao.java:73]: This potentially looks like a good candidate for extracting up into an abstract class that all of these DAO tests can inherit? Is that possible?

File: service/src/test/java/bio/terra/cbas/dao/TestMethodDao.java [hunk 3]
```diff
@@ -50,22 +58,40 @@ class TestMethodDao {
...
+
   @BeforeAll
-  void setUp() {
+  static void setup() {
+    postgres.start();
```
Comment (cjllanwarne) [inline service/src/test/java/bio/terra/cbas/dao/TestMethodDao.java:77]: This means that each test reuses the same test container as the previous one... is that deliberate? I guess it's more efficient if the tests never interfere with each other but feels potentially risky?

File: service/src/test/java/bio/terra/cbas/dao/TestMethodDao.java [hunk 4]
```diff
@@ -50,22 +58,40 @@ class TestMethodDao {
...
-    int recordsDeleted2 = methodDao.deleteMethod(methodId2);
-
-    assertEquals(1, recordsDeleted1);
-    assertEquals(1, recordsDeleted2);
+  @AfterEach
+  void cleanupDb() throws SQLException {
+    DriverManager.getConnection(
+            postgres.getJdbcUrl(), postgres.getUsername(), postgres.getPassword())
+        .createStatement()
+        .execute("DELETE FROM run_set; DELETE FROM method_version; DELETE FROM method;");
```
Comment (cjllanwarne) [inline service/src/test/java/bio/terra/cbas/dao/TestMethodDao.java:94]: This might also be good to extract up into an abstract super class?

Comment (cjllanwarne) [inline service/src/test/java/bio/terra/cbas/dao/TestMethodDao.java:94]: Should we also have `DELETE FROM run`?

Comment (cjllanwarne) [inline service/src/test/java/bio/terra/cbas/dao/TestMethodDao.java:94]: And github method details?

Comment (mspector) [inline service/src/test/java/bio/terra/cbas/dao/TestMethodDao.java:94]: make sure to delete github method details and any other tables that I've missed here

File: service/src/test/java/bio/terra/cbas/dao/TestMethodDao.java (hunk 3 above)
Comment (cjllanwarne) [reply service/src/test/java/bio/terra/cbas/dao/TestMethodDao.java:77]: Never mind - the aftereach makes sure they are all independent!

File: service/src/test/java/bio/terra/cbas/dao/TestMethodDao.java (hunk 4 above)
Comment (mspector) [reply service/src/test/java/bio/terra/cbas/dao/TestMethodDao.java:94]: I typed my previous comment before seeing this one -- yes!

**Comment by mspector** [comment]
closing this PR in favor of https://github.com/DataBiosphere/cbas/pull/261

**Pull request closed without merging**